pydicom/valuerep.py

```diff
@@ -43,6 +43,312 @@
 # (the component separator '=' is handled separately)
 PN_DELIMS = {0x5e}
 
+# maximum allowed value length for string VRs
+# VRs with a maximum length of 2^32 (UR and UT) are not checked
+MAX_VALUE_LEN = {
+    "AE": 16,
+    "CS": 16,
+    "DS": 16,
+    "IS": 12,
+    "LO": 64,
+    "LT": 10240,
+    "SH": 16,
+    "ST": 1024,
+    "UI": 64
+}
+
+
+def _range_regex(regex: str) -> str:
+    """Compose a regex that allows ranges of the given regex,
+    as defined for VRs DA, DT and TM in PS 3.4, C.2.2.2.5.
+    """
+    return fr"^{regex}$|^\-{regex} ?$|^{regex}\- ?$|^{regex}\-{regex} ?$"
+
+
+# regular expressions to match valid values for some VRs
+AE_REGEX = re.compile(r"^[\x20-\x7e]*$")
```

Comment:
Doesn't this allow backslash?

Reply:
The validation is done for single values, e. g. after the split by backslash, so this is not needed here.

Reply:
But I mean, this is the value setter check, so if someone uses a value of "MY\AE" then it'll pass validation but end up as VM 2. Or have I missed something?